core/src/test/scala/unit/kafka/server/CreateTopicsRequestTest.scala

```diff
@@ -19,76 +19,61 @@ package kafka.server
-    val config3 = Map("min.insync.replicas" -> "2").asJava
-    validateValidCreateTopicsRequests(new CreateTopicsRequest.Builder(Map("topic3" -> new CreateTopicsRequest.TopicDetails(5, 2.toShort, config3)).asJava, timeout).build())
+    validateValidCreateTopicsRequests(topicsReq(Seq(topicReq("topic1"))))
+    validateValidCreateTopicsRequests(topicsReq(Seq(topicReq("topic2", replicationFactor = 3))))
+    validateValidCreateTopicsRequests(topicsReq(Seq(topicReq("topic3",
+      numPartitions = 5, replicationFactor = 2, config = Map("min.insync.replicas" -> "2")))))
     // Manual assignments
-    val assignments4 = replicaAssignmentToJava(Map(0 -> List(0)))
-    validateValidCreateTopicsRequests(new CreateTopicsRequest.Builder(Map("topic4" -> new CreateTopicsRequest.TopicDetails(assignments4)).asJava, timeout).build())
-    val assignments5 = replicaAssignmentToJava(Map(0 -> List(0, 1), 1 -> List(1, 0), 2 -> List(1, 2)))
-    val config5 = Map("min.insync.replicas" -> "2").asJava
-    validateValidCreateTopicsRequests(new CreateTopicsRequest.Builder(Map("topic5" -> new CreateTopicsRequest.TopicDetails(assignments5, config5)).asJava, timeout).build())
+    validateValidCreateTopicsRequests(topicsReq(Seq(topicReq("topic4", assignment = Map(0 -> List(0))))))
+    validateValidCreateTopicsRequests(topicsReq(Seq(topicReq("topic5",
+      assignment = Map(0 -> List(0, 1), 1 -> List(1, 0), 2 -> List(1, 2)),
+      config = Map("min.insync.replicas" -> "2")))))
     // Mixed
-    val assignments8 = replicaAssignmentToJava(Map(0 -> List(0, 1), 1 -> List(1, 0), 2 -> List(1, 2)))
-    validateValidCreateTopicsRequests(new CreateTopicsRequest.Builder(Map(
-      "topic6" -> new CreateTopicsRequest.TopicDetails(1, 1.toShort),
-      "topic7" -> new CreateTopicsRequest.TopicDetails(5, 2.toShort),
-      "topic8" -> new CreateTopicsRequest.TopicDetails(assignments8)).asJava, timeout).build()
-    )
-    validateValidCreateTopicsRequests(new CreateTopicsRequest.Builder(Map(
-      "topic9" -> new CreateTopicsRequest.TopicDetails(1, 1.toShort),
-      "topic10" -> new CreateTopicsRequest.TopicDetails(5, 2.toShort),
-      "topic11" -> new CreateTopicsRequest.TopicDetails(assignments8)).asJava, timeout, true).build()
-    )
+    validateValidCreateTopicsRequests(topicsReq(Seq(topicReq("topic6"),
+      topicReq("topic7", numPartitions = 5, replicationFactor = 2),
+      topicReq("topic8", assignment = Map(0 -> List(0, 1), 1 -> List(1, 0), 2 -> List(1, 2))))))
+    validateValidCreateTopicsRequests(topicsReq(Seq(topicReq("topic9"),
+      topicReq("topic10", numPartitions = 5, replicationFactor = 2),
+      topicReq("topic11", assignment = Map(0 -> List(0, 1), 1 -> List(1, 0), 2 -> List(1, 2)))),
+      validateOnly = true))
   }
 
   @Test
   def testErrorCreateTopicsRequests() {
-    val timeout = 10000
```

Comment:
Should we keep this timeout since the default timeout is 0 and could fail a request prematurely?

Reply:
Good point.  Let's set the default timeout to something like a minute, so that we don't get bogus failures in unit tests.  (This shouldn't matter for AdminClient since AC always explicitly sets the timeout.)